src/book/lib/transaction/createTransaction.ts

```diff
@@ -0,0 +1,99 @@
+import { DateTime } from 'luxon';
+import { toAmountWithScale } from '@/helpers/number';
+import Stocker from '@/apis/Stocker';
+
+import {
+  Commodity,
+  Price,
+  Split,
+  Transaction,
+} from '@/book/entities';
+import { getMainCurrency } from '@/book/queries';
+import { isInvestment, isAsset } from '@/book/helpers';
+
+export async function createTransaction(
+  when: DateTime,
+  description: string,
+  mainSplit: Split,
+  splits: Split[],
+): Promise<void> {
+  const { currency, rate } = getTransactionCurrency(mainSplit, splits);
+
+  const transaction = Transaction.create({
+    guid: crypto.randomUUID().substring(0, 31),
+    fk_currency: currency,
+    date: when,
+    description,
+  });
+
+  splits.forEach(split => {
+    split.fk_transaction = transaction;
+
+    // If we've exchanged currency to avoid having STOCK currency
+    // as currency transaction, correct value and quantity
+    if (split.account.commodity.guid === currency.guid) {
+      split.valueNum = split.quantityNum;
+      split.valueDenom = split.quantityDenom;
+    }
+  });
+
+  // eslint-disable-next-line no-param-reassign
+  mainSplit.fk_transaction = transaction;
+
+  // We recalculate value for mainSplit in case we've swapped currencies
+  const { amount, scale } = toAmountWithScale(mainSplit.value * rate);
+  mainSplit.valueNum = amount;
+  mainSplit.valueDenom = parseInt('1'.padEnd(scale + 1, '0'), 10);
+
+  const mainCurrency = await getMainCurrency();
+  if (currency.guid !== mainCurrency.guid && isInvestment(mainSplit.account)) {
```

Comment:
For investment transactions, we want to add a price instance to the DB so we know the value of the transaction in the main currency when representing it in the investments page.